### DISCUSSION TITLE
clearNuxtData function to clear useAsyncData data

### DISCUSSION THREAD

**Discussion by jf908** [nuxt/nuxt#16548] [Ideas]
I want to clear all of the data cached by `useAsyncData` when a user logs out.

I expected an alternative to `refreshNuxtData` that clears the data instead of refreshing it but there does not seem to be one (unless I'm missing something).

**Comment by jf908** [comment]
Looks like [clearNuxtData](https://v3.nuxtjs.org/api/utils/clear-nuxt-data) exists now, great!

**Comment by develth** [comment]
Hey @jf908 

how you work with that?
When i do a clearNuxtData() and then logout the user, data is still there. Only Full Page reload helps. 
Thanks!

**Reply by jf908** [reply]
clearNuxtData does what it says in the docs, it clears data that is cached via useAsyncData. If it does not then you’ll have to report an issue to Nuxt. clearNuxtData won’t clear cookies or any separate data structures and you have to deal with those manually.

**Reply by develth** [reply]
yes, it clears data. But after `refreshNuxtData` they are back.  Or within dev mode if i do a code update for example. 
Is that the wanted behaviour? If not i will go in deeper and create in issue

**Reply by jf908** [reply]
refreshNuxtData will re-run useAsyncData but if’s returning cached data rather than fetching it from scratch that definitely seems like a bug yes

**Reply by heikopaiko** [reply]
any updates on that?

**Reply by jf908** [reply]
Haven't used this function recently so I don't know - if it doesn't work I suggest making a minimal repo and creating an issue

**Comment by maximelebreton** [comment]
Same problem than @develth.
When I use `clearNuxtData('myItem')`, and then reload the page (who use `useFetch(''/api/my-item", {key: 'myItem'})`, the old data is still loaded...
I need to enable disable cache on chrome devltools to have a fresh version...!
If it's not a bug, this is a really unexcepted behaviour.

Actually it's very undocumented and unclear so it's really difficult to have a good understanding of how invalidating cache works on Nuxt (if we want to invalidate cache, there is no way to do this for client and for server at the same time), even more if there is some bugs.

I'll try to create a minimal repo soon to isolate it.
